### DISCUSSION TITLE
Not showing Port while trying to use attiny85 (I'm a begginer)

### DISCUSSION THREAD

**Discussion by BatoviJR** [SpenceKonde/ATTinyCore#879] [Support / Q & A]
Hello everyone, I've recently been trying to program my digispark attiny85 on my Linux (Debian based). I installed Micronucleus (and put the board "ATtiny85(Micronucleus/DigiSpark)" ), but the Arduino IDE doesn't show the port option (It doesn't even let me click to see the options). So I did some research and the only thing I could find out is that I need to install the drivers so that they recognize my port. But the digistump drivers are no longer available and I don't know any other way to do it.

*When I try to upload some code to my attiny85, Micronucleus runs 100%, but it doesn't seem to work (my attiny doesn't do anything). I think it's because I didn't put the port.

I'm a total beginner in this subject, I understand that it may be a silly question, but I couldn't find anything that would help me with this problem.

I would be extremely grateful if someone could help me.
Thanks.

![The "port" not appearing in tools](https://github.com/user-attachments/assets/e37ff529-0461-437a-99f7-b6b4440ee47b)

![The Micronucleus complete](https://github.com/user-attachments/assets/6dfb305e-4548-47e8-a212-e7170978b4e2)

![My attiny85 plugged](https://github.com/user-attachments/assets/1042c18f-5a5d-430d-87c0-27d454a4f3d2)

**Comment by tnsolley** [comment]
I've also just bought a Digispark board, the driver install via Arduino worked but the object isn't appearing as a COM item for me either. Under Device manager it's listed as "libusb-win32 devices -> Digispark Bootloader".
What should I be doing with this board that I haven't already? I've attempted connection with Arduino IDE 1.8.9 and 2.3.4, same result (no appearance as COM object).
Edit:
I've also just plugged the Digispark directly in to my PC via USB-port like the image above, don't quite understand the "ISP" page under the Programming Guide page (do I need some special sketch, and another Arduino, to burn the bootloader first to the Digispark before it will appear as a COM object?)

**Comment by hmeijdam** [comment]
The digispark bootloader nor the Micronucleus bootloader that comes with Attinycore show as a COM port 

After burning Micronucleus onto the attiny85, which you need an **[Arduino as ISP](https://docs.arduino.cc/built-in-examples/arduino-isp/ArduinoISP/)** for it shows up as "Unknown Device" (at least on my PC).

![image](https://github.com/user-attachments/assets/f3d20898-3160-4077-b873-80eafa7d4d5c)

**Reply by tnsolley** [reply]
After burning Micronucleus onto the Digispark, are you able to select it as a COM-port object in the Arduino IDE?

**Reply by hmeijdam** [reply]
I have never seen that. 
These bootloaders do not work with an upload method that involves a serial COM port. 
You press "upload" and when told to connect the digispark, only then connect it.
Connecting it too soon will make the upload fail, as the bootloader is waiting only for a very brief moment until moving on to the code.

**Comment by tnsolley** [comment]
Does this mean that one cannot use the Serial Monitor to view output from
the Digispark?

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Reply by hmeijdam** [reply]
Correct. You would need a separate USB<>serial adapter or find some software that emulates a USB COM port via the USB connector.
